File: python/l_list.py
class Node:

    def __init__(self, data, nextNode):
        if data is not None:
            self.data = data
        else:
             self.data = None

        if nextNode is not None:
            self.nextNode = nextNode
        else:
            self.nextNode = None

    def appendToTail(self, node):
        nodeToAppend = node
        tailNode = self

        while tailNode.nextNode is not None:
            tailNode = tailNode.nextNode

        tailNode.nextNode = nodeToAppend

    def printNodes(self):
        currentNode = self

        while currentNode is not None:

            if currentNode.nextNode is not None:
                print "Node: %s ->" % currentNode.data,
            else:
                print "Node: %s" % currentNode.data

            currentNode = currentNode.nextNode


das_list = Node("wunderbar", None)

das_list.printNodes()

yo = Node("yo", None)

das_list.appendToTail(yo)

das_list.printNodes()

File: ruby/l_list.rb
class Node
  def initialize(data = nil, next_node = nil)
    @next_node = next_node
    @data = data
  end

  def appendToTail(node)
    node_to_append = node
    tail_node = self

    until tail_node.get_next_node == nil
      tail_node = tail_node.get_next_node
    end

    tail_node.set_next_node(node_to_append)
  end

  def get_next_node
    return @next_node
  end

  def set_next_node(new_node)
    @next_node = new_node
  end

  def get_data
    return @data
  end

  def printNodes
    current_node = self

    until current_node == nil
      current_node.get_next_node == nil ? print("Node: #{current_node.get_data} \n") : print("Node: #{current_node.get_data} ->")
      current_node = current_node.get_next_node
    end

  end

end

# das_list = Node.new("sup")
# das_list.printNodes()
#
# yo = Node.new("yo")
# das_list.appendToTail(yo)
#
# das_list.printNodes()
